onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -269,10 +269,32 @@ def aten_angle(self: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_any(self: TensorType) -> TensorType:
+@torch_op("aten::any", trace_only=True)
+def aten_any(self: TTensor, dim: Optional[int] = None, keepdim: bool = True) -> BOOL:
```

Comment:
I think keepdim should default to False, or set as `keepdim: Optional[bool] = None`, and explicitly set to `False` if `None`.

That way we can avoid hard coding `keepdims=0` later.

https://github.com/microsoft/onnx-script/blob/ec7c26471e0ef1896ba492549f33d0367b0301e4/onnxscript/function_libs/torch_aten/ops/core.py#L292

Reply:
Shouldn't the default value be determined by the spec of the aten op?

Reply:
Looking at it again, I figured we are supporting 2 overloads of `aten::any` with this function. So for `- func: any(Tensor self) -> Tensor` we need actually figuring out what the behavior is since both `dim` and `keepdim` are not provided. So I think this is fine.

And for `- func: any.dim(Tensor self, int dim, bool keepdim=False) -> Tensor`, yea you are right, the current default is wrong and it should be False.

```
- func: any(Tensor self) -> Tensor
  device_check: NoCheck   # TensorIterator
  structured_delegate: any.all_out
  variants: method, function
  dispatch:
    SparseCPU, SparseCUDA: any_sparse

- func: any.dim(Tensor self, int dim, bool keepdim=False) -> Tensor
  device_check: NoCheck   # TensorIterator
  structured_delegate: any.out
  variants: function, method
```

Reply:
cc @justinchuby it is weird the wrong default is not caught in op test, I wonder if it is just not enough coverage in original opinfo or something else.

Reply:
I think we should not have combined the two overloads, and yes - you have a great point that we should verify the opinfo to assert how much confidence it can provide us for each op before trusting it. I will add this to the authoring guide